src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewComponentResult.cs

```diff
@@ -47,17 +37,32 @@ public class ViewComponentResult : ActionResult
         public Type ViewComponentType { get; set; }
 
         /// <summary>
+        /// Get the view data model.
+        /// </summary>
+        public object Model => ViewData?.Model;
+
+        /// <summary>
         /// Gets or sets the <see cref="ViewDataDictionary"/> for this result.
         /// </summary>
         public ViewDataDictionary ViewData { get; set; }
 
         /// <summary>
+        /// Gets or sets the <see cref="ITempDataDictionary"/> for this result.
+        /// </summary>
+        public ITempDataDictionary TempData { get; set; }
```

Comment:
I moved the properties in ViewComponentResult to be consistent with the orderings in ViewResult/PartialViewResult. Should I order them all alphabetically?